sklearn_numba_dpex/common/kernels.py

```diff
@@ -228,31 +235,57 @@ def make_sum_reduction_2d_axis1_kernel(
 
     Notes
     -----
-    `work_group_size` is assumed to be a power of 2.
+    if `size1` is None then the kernel expects 1d tensor inputs, and `axis` parameter
+    is ignored.
 
-    if `size1` is None then the kernel expects 1d tensor inputs. If `size1` is not None
-    then the expected shape of input tensors is `(size0, size1)`, and the reduction
-    operation is equivalent to input.sum(axis=1). Depending on the size of the sum
-    axis, it might be worth tuning `work_group_size` for better performance.
+    If `size1` is not None then the expected shape of input tensors is
+    `(size0, size1)`, and the reduction operation is equivalent to
+    `input.sum(axis=axis)`.
+
+    if `size1` is None or if `axis` is `0`, then `work_group_size` is assumed to be a
+    power of 2, and the parameter `sub_group_size` is ignored.
+
+    if `size1` is not `None` and `axis` is `1`, then `work_group_size` is assumed to be
+    a multiple of `sub_group_size`, such that `work_group_size // sub_group_size` is
+    a power of 2.
+
+    Depending on the size of the sum axis, it might be worth tuning `work_group_size`
+    for better performance. When `axis` is `1`, tuning `sub_group_size` with respect
+    to the size of the other axis might also be beneficial.
```

Comment:
`fused_unary_func` is already documented in one of the blocks above `Notes`. OK for renaming.

```
    If `fused_elementwise_func` is not None, it will be applied element-wise before 
    summing. It must be a function that will be interpreted as a dpex.func and is
    subject to the same rules. It is expected to take one scalar argument and returning
    one scalar value. lambda functions are advised against since the cache will not 
    work with lambda functions. sklearn_numba_dpex.common._utils expose some 
    pre-defined `fused_elementwise_func`s.
```

Adding memory layouts expectations:

>     This implementation is optimized for C-contiguous arrays.